python/lsst/daf/butler/script/queryDataIds.py

```diff
@@ -78,6 +82,26 @@ def queryDataIds(repo, dimensions, datasets, where, collections, order_by, limit
     # Docstring for supported parameters is the same as Registry.queryDataIds
 
     butler = Butler(repo)
+
+    if not dimensions and datasets and collections:
+        # Determine the dimensions relevant to all given dataset types.
+        graph = None
+        dataset_types = list(butler.registry.queryDatasetTypes(datasets))
+        for dataset_type in dataset_types:
+            if graph is None:
```

Comment:
`graph` here is a `set`, no? Maybe assign it to an empty set above, as that would make the logic of this loop more obvious.

Reply:
If I start with an empty set then ANDing anything will result in an empty set. I use `None` to indicate that nothing has been looked at yet. The alternative would be to build a default set with all possible dimensions in it. `None` seemed easier.